include/tvm/relay/transform.h

```diff
@@ -377,6 +377,15 @@ TVM_DLL Function InferType(const Function& f,
                            const IRModule& mod,
                            const GlobalVar& var);
 
+/*!
+ * \brief Infer the type of an expression base on it's inputs.
+ *
+ * \param expr the Expr.
+ *
+ * \return A type checked Expr with its checked_type field populated.
+ */
+TVM_DLL Expr InferType(const Expr& expr);
```

Comment:
I am not sure we need this Expr->Expr helper. You actually only use it once. I believe we can achieve this using a one-liner as well, i.e. something like this:`expr = IRModule::FromExpr(expr)->Lookup("main")`.

Reply:
I turned it into a utility based on @masahi's request, thoughts?

Reply:
transform should only contains IRModule->IRModule transformations, if we need a util that is based on this, let us put it as an internal function for now

Reply:
Reverted the commit that moved this into the header.